### PR TITLE
Base setup for Chef Solo (#1011)

### PR DESCRIPTION
**Pull request by sethetter**
This includes the necessary configuration setup, make tasks, etc, for provisioning a server using Chef Solo. It uses our cookbooks for a Madison server, which are still in progress, but is agnostic to how those get run (unless we add more cookbooks).

I'll be opening a new issue to tackle full deployment documentation using Chef Solo and Capistrano after this.

@doshitan @krues8dr

### CODE REVIEW

File: config/chef/data_bags/users/deploy.json.example [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "comment": "Deploy user",
+  "shell": "/bin/bash",
+
+  // TODO: Include instructions on encrypting a password
+  "password": "REPLACE ME",
+
+  "ssh_keys": [
+    "REPLACE ME"
+  ]
+}
```
Comment (krues8dr) [inline config/chef/data_bags/users/deploy.json.example]: If this is using the `opengov-users` cookbook, I think `deploy` is gonna get stepped on by the `deploy` user that that sets up.  I'd recommend using a different default user name here?

**Comment by krues8dr** [comment]
1 comment, the rest looks good.

File: config/chef/data_bags/users/deploy.json.example (hunk 1 above)
Comment (sethetter) [inline config/chef/data_bags/users/deploy.json.example]: That's corrected in #1029 [here](https://github.com/opengovfoundation/madison/blob/6666f780d3c96ec0b02d08f23aeedc34a5f38a95/config/chef/data_bags/users/you.json.example)

Comment (krues8dr) [inline config/chef/data_bags/users/deploy.json.example]: :+1:

**Pull request merged**